### ISSUE TITLE
Correct indentation not applied to `when` expressions with trailing commas enabled 

### ISSUE THREAD

**Issue by jkmckay** [detekt/detekt#5174]
## Expected Behavior

When running detekt with both trailing commas and indentation enabled, i.e.
```yaml
  Indentation:
    active: true
    autoCorrect: true
    indentSize: 4
    continuationIndentSize: 4
/* ... */
  TrailingComma:
    active: true
    autoCorrect: true
    allowTrailingComma: true
    allowTrailingCommaOnCallSite: true
```
Trailing commas should be added as appropriate _and_ indentation should be applied.

For example something like this:
```kotlin

enum class TestEnum { EXAMPLE_A, EXAMPLE_B, EXAMPLE_C }

fun exampleFunction(enum: TestEnum) {
    when (enum) {
        TestEnum.EXAMPLE_A,
        TestEnum.EXAMPLE_B -> enum.toString()
        else -> {}
    }
}
```
should be formatted as this:
```kotlin
enum class TestEnum { EXAMPLE_A, EXAMPLE_B, EXAMPLE_C }

fun exampleFunction(enum: TestEnum) {
    when (enum) {
        TestEnum.EXAMPLE_A,
        TestEnum.EXAMPLE_B,
        -> enum.toString()
        else -> {}
    }
}
```

## Observed Behavior
What happens is this:
(**until detekt is ran for a second time and then the expected output is produced.**)
```kotlin
enum class TestEnum { EXAMPLE_A, EXAMPLE_B, EXAMPLE_C }

fun exampleFunction(enum: TestEnum) {
    when (enum) {
        TestEnum.EXAMPLE_A,
        TestEnum.EXAMPLE_B,
-> enum.toString()
        else -> {}
    }
}
```
## Steps to Reproduce
Enable trailing commas and observe `when` expresssions formatting incorrectly  on first run of detekt.

## Context
 Preventing team from agreeing to enable the trailing commas option and manually formatting.

## Your Environment
Tested on Detekt 1.19.0 & 1.21.0

**Comment by schalkms** [comment]
Both mentioned rules are **native** ktlint rules. Detekt just provides a wrapper around them.
Could you please try to reproduce this with **plain** ktlint and potentially report to this to their repo?

**Comment by jkmckay** [comment]
Thanks, I've checked the Ktlint repo and an open issue for this already exists. Will close this one.
